### PR TITLE
Use multiple values in smart variables

### PR DESCRIPTION
**Pull request by pedrocarmona**
Hi @ankane,

This PR allows the user to select multiple values from a smart_variable.

Its possible to use smart_variables in queries with IN.
```
SELECT ...
FROM ...
WHERE user_id IN({user_ids})
```

### CODE REVIEW

File: app/controllers/blazer/base_controller.rb [hunk 1]
```diff
@@ -33,27 +33,35 @@ def process_vars(statement, data_source)
...
+                  value = value.to_i
+                elsif value =~ /\A\d+\.\d+\z/
+                  value = value.to_f
+                end
+              end
+              value = ActiveRecord::Base.connection.quote(value)
+              if values.last == value_str
+                statement.gsub!("{#{var}}", value)
+              else
+                statement.gsub!("{#{var}}", "#{value}, {#{var}}")
```
Comment (pedrocarmona) [inline app/controllers/blazer/base_controller.rb]: Should this be also activated only for plural variables?

File: app/assets/javascripts/blazer/selectize.js [hunk 2]
```diff
@@ -2707,7 +2707,7 @@
           label = self.options[self.items[i]][self.settings.labelField] || '';
           options.push('<option value="' + escape_html(self.items[i]) + '" selected="selected">' + escape_html(label) + '</option>');
         }
-        if (!options.length && !this.$input.attr('multiple')) {
+        if (!options.length) {
```
Comment (pedrocarmona) [inline app/assets/javascripts/blazer/selectize.js]: Either were, or in controller append null to all missing smart variables

File: app/views/blazer/_variables.html.erb [hunk 3]
```diff
@@ -8,27 +8,29 @@
...
-            create: true
-          });
-        </script>
-      <% else %>
-        <%= text_field_tag var, params[var], style: "width: 120px; margin-right: 20px;", autofocus: i == 0 && !var.end_with?("_at") && !params[var], class: "form-control" %>
-        <% if var.end_with?("_at") %>
+      <div class="form-group">
+        <%= label_tag var, var %>
+        <% if (data = @smart_vars[var]) %>
+          <%= select_tag var, options_for_select([[nil, nil]] + data, selected: params[var]), style: "margin-right: 20px; width: 200px; display: none;", multiple: true %>
```
Comment (pedrocarmona) [inline app/views/blazer/_variables.html.erb]: multiple: true could be conditional and based in regex from query or if variable is in plural (```user_ids```)

**Comment by pedrocarmona** [comment]
@ankane, can you review?

**Comment by bernstein7** [comment]
@pedrocarmona hey, why it was closed?

**Comment by pedrocarmona** [comment]
it was a bit complex, I stopped enforcing a use case to it, but if you want I can restore the branch and we can test and improve together

**Pull request closed without merging**